org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/projectWizard/TemplateProjectWizardFragment.xtend

```diff
@@ -0,0 +1,303 @@
+					name.enabled = advanced.value
+					path.enabled = advanced.value
+					if (!advanced.value) {
+						name.value = "Xtext"
+						path.value = "�language.fileExtensions.get(0)�"
+					}
+				}
+			
+				override protected validate() {
+					if (path.value.matches('[a-z][a-z0-9_]*(/[a-z][a-z0-9_]*)*'))
+						null
+					else
+						new Status(ERROR, "Wizard", "'" + path + "' is not a valid package name")
+				}
+			
+				�IF pluginProject�
+					override generateProjects(IProjectGenerator generator) {
+						generator.generate(
+							new PluginProjectFactory().setProjectName(getProjectInfo().getProjectName()).setLocation(
+								getProjectInfo().getLocationPath()).addProjectNatures(JavaCore.NATURE_ID,
+								"org.eclipse.pde.PluginNature", XtextProjectHelper.NATURE_ID).addBuilderIds(JavaCore.BUILDER_ID).
+								addFolders(Arrays.asList("src")).addFile(�quotes�src/�openVar�path�closeVar�/Model.�language.fileExtensions.get(0)��quotes�, �quotes�
+									/*
+									 * This is an example model
+									 */
+									Hello �openVar�name�closeVar�!
+								�quotes�))
+					}
+				�ELSE�
+					override generateProjects(IProjectGenerator generator) {
+						generator.generate(
+							new ProjectFactory().setProjectName(getProjectInfo().getProjectName()).setLocation(
+								getProjectInfo().getLocationPath()).addProjectNatures(XtextProjectHelper.NATURE_ID).
+								addFile(�quotes�src/�openVar�path�closeVar�/Model.�language.fileExtensions.get(0)��quotes�, �quotes�
+									/*
+									 * This is an example model
+									 */
+									Hello �openVar�name�closeVar�!
+								�quotes�))
+					}
+				�ENDIF�
+			}
+		'''
+		file.writeTo(projectConfig.eclipsePlugin.src)
+	}
+
+	def generateNewProjectWizard() {
+		val genClass = getProjectWizardClassName.typeRef
+
+		val file = fileAccessFactory.createGeneratedJavaFile(genClass)
+		file.content = '''
+			public class �genClass.simpleName� extends �"org.eclipse.xtext.ui.wizard.template.TemplateNewProjectWizard".typeRef� {
+				@Override
+				protected String getGrammarName() {
+					return "�grammar.name�";
+				}
+			}
+		'''
+		file.writeTo(projectConfig.eclipsePlugin.src)
+	}
+
+	def generateDefaultIcons() {
+		val projectTemplate = fileAccessFactory.createBinaryFile("project_template.png")
+		projectTemplate.content = readBinaryFileFromPackage("project_template.png")
+		projectTemplate.writeTo(projectConfig.eclipsePlugin.icons)
+
+		val newProject = fileAccessFactory.createBinaryFile("new_" + grammar.simpleName + "_proj.gif")
+		newProject.content = readBinaryFileFromPackage("new_xproj.gif")
+		newProject.writeTo(projectConfig.eclipsePlugin.icons)
+	}
+
+	private def byte[] readBinaryFileFromPackage(String fileName) {
+		// TODO use "readAllBytes" as soon as we switch to Java 9
+		// return TemplateProjectWizardFragment.getResourceAsStream(fileName).readAllBytes();
+		val bis = new BufferedInputStream(TemplateProjectWizardFragment.getResourceAsStream(fileName));
```

Comment:
Beware unclosed streams. I recommend using Guava utilities for these operations.